python/tvm/contrib/cudnn.py

```diff
@@ -384,3 +384,288 @@ def conv2d_forward(x,
+                     stride_w,
+                     dilation_d,
+                     dilation_h,
+                     dilation_w,
+                     x_shape,
+                     w_shape,
+                     y_shape):
+    """Choose the best algo for the given input.
+
+    Paramters
+    ---------
+    tensor_format: int
+        0: CUDNN_TENSOR_NCHW
+        1: CUDNN_TENSOR_NHWC
+        2: CUDNN_TENSOR_NCHW_VECT_C
+    pad_d: int
+        depth pad
+    pad_h: int
+        height pad
+    pad_w: int
+        weight pad
+    stride_d: int
+        depth stride
+    stride_h: int
+        height stride
+    stride_w: int
+        width stride
+    dilation_d: int
+        depth dilation
+    dilation_h: int
+        height dilation
+    dilation_w: int
+        width dilation
+    x_shape: list
+        input shape
+    w_shape: list
+        weight shape
+    y_shape: list
+        output shape
+
+    Returns
+    -------
+    algo: int
+        algo chosen by CUDNN
+    """
+    func = _get_global_func("tvm.contrib.cudnn.conv3d.find_algo")
+    return func(tensor_format,
+                pad_d,
+                pad_h,
+                pad_w,
+                stride_d,
+                stride_h,
+                stride_w,
+                dilation_d,
+                dilation_h,
+                dilation_w,
+                x_shape[0].value,
+                x_shape[1].value,
+                x_shape[2].value,
+                x_shape[3].value,
+                x_shape[4].value,
+                w_shape[0].value,
+                w_shape[1].value,
+                w_shape[2].value,
+                w_shape[3].value,
+                w_shape[4].value,
+                int(y_shape[0]),
+                int(y_shape[1]),
+                int(y_shape[2]),
+                int(y_shape[3]),
+                int(y_shape[4]))
+
+
+def conv3d_forward(x,
+                   w,
+                   stride_d=1,
+                   stride_h=1,
+                   stride_w=1,
+                   pad_d=0,
+                   pad_h=0,
+                   pad_w=0,
+                   dilation_d=1,
+                   dilation_h=1,
+                   dilation_w=1,
+                   conv_mode=1,
+                   tensor_format=0,
+                   algo=-1):
+    """Create an extern op that compute 2D convolution with CuDNN
+
+    Parameters
+    ----------
+    x: Tensor
+        input feature map
+    w: Tensor
+        convolution weight
+    stride_d: int
+        depth stride
+    stride_h: int
+        height stride
+    stride_w: int
+        width stride
+    pad_d: int
+        depth pad
+    pad_h: int
+        height pad
+    pad_w: int
+        weight pad
+    dilation_d: int
+        depth dilation
+    dilation_h: int
+        height dilation
+    dilation_w: int
+        width dilation
+    conv_mode: int
+        0: CUDNN_CONVOLUTION
+        1: CUDNN_CROSS_CORRELATION
+    tensor_format: int
+        0: CUDNN_TENSOR_NCHW
```

Comment:
Is this value compatible with 3D convolution?

Reply:
Yes. cudnn doesn't have specific tensor format for 3D and up dimentions convolution, the leading format is same "NCHW" and the additional dimentions would be append to the tail for 'channel first' format.  Please see below citation from nvidia [DL SDK](https://docs.nvidia.com/deeplearning/sdk/cudnn-api/index.html#cudnnSetFilterNdDescriptor)
```
format

    Input.Type of the filter layout format. If this input is set to CUDNN_TENSOR_NCHW, which is one of the enumerant values allowed by cudnnTensorFormat_t descriptor, then the layout of the filter is as follows:

        For N=4, a 4D filter descriptor, the filter layout is in the form of KCRS:
            K represents the number of output feature maps
            C is the number of input feature maps
            R is the number of rows per filter
            S is the number of columns per filter
        For N=3, a 3D filter descriptor, the number S (number of columns per filter) is omitted.
        For N=5 and greater, the layout of the higher dimensions immediately follow RS.

    On the other hand, if this input is set to CUDNN_TENSOR_NHWC, then the layout of the filter is as follows:

        For N=4, a 4D filter descriptor, the filter layout is in the form of KRSC.
        For N=3, a 3D filter descriptor, the number S (number of columns per filter) is omitted and the layout of C immediately follows R.
        For N=5 and greater, the layout of the higher dimensions are inserted between S and C. For more information, see cudnnTensorFormat_t.
```